test/support/otlp_endpoint_test.rb

```diff
@@ -0,0 +1,287 @@
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil))
+      assert_entity_endpoint_nil
+    end
+
+    # 10 lambda doesn't care about SW_APM_SERVICE_KEY and OTEL_EXPORTER_OTLP_ENDPOINT
+    it 'SW_APM_API_TOKEN invalid inside lambda' do
+      ENV['SW_APM_API_TOKEN'] = nil
+      ENV['LAMBDA_TASK_ROOT'] = '/task/vartask'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token('METRICS')
+
+      assert_nil(endpoint.instance_variable_get(:@token))
+      assert_nil(endpoint.instance_variable_get(:@service_name))
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal false
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil))
+      assert_entity_endpoint_nil
+    end
+
+    it 'SW_APM_API_TOKEN valid inside lambda' do
+      ENV['SW_APM_API_TOKEN'] = '0123456789abcde0123456789abcde0123456789abcde0123456789abcde1234'
+      ENV['LAMBDA_TASK_ROOT'] = '/task/vartask'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint.config_token('METRICS')
+
+      assert_nil(endpoint.instance_variable_get(:@token))
+      assert_nil(endpoint.instance_variable_get(:@service_name))
+      _(endpoint.instance_variable_get(:@agent_enable)).must_equal true
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_HEADERS', nil))
+      assert_entity_endpoint_nil
+    end
+  end
+
+  describe 'config_endpoint' do
+    let(:endpoint_types) { %w[TRACES METRICS LOGS] }
+
+    it 'no OTEL ENDPOINT and no SW_APM_COLLECTOR' do
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint_types.each { |data_type| endpoint.configure_otlp_endpoint(data_type) }
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil))
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443/v1/traces'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443/v1/metrics'
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_ENDPOINT', nil)).must_equal 'https://otel.collector.na-01.cloud.solarwinds.com:443/v1/logs'
+      _(ENV.fetch('SW_APM_COLLECTOR', nil)).must_equal 'apm.collector.na-01.cloud.solarwinds.com'
+    end
+
+    it 'OTEL ENDPOINT to local and no SW_APM_COLLECTOR' do
+      ENV['OTEL_EXPORTER_OTLP_ENDPOINT'] = 'http://localhost:4317'
+      endpoint = SolarWindsAPM::OTLPEndPoint.new
+      endpoint_types.each { |data_type| endpoint.configure_otlp_endpoint(data_type) }
+
+      _(ENV.fetch('OTEL_EXPORTER_OTLP_ENDPOINT', nil)).must_equal 'http://localhost:4317'
+
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_TRACES_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_METRICS_ENDPOINT', nil))
+      assert_nil(ENV.fetch('OTEL_EXPORTER_OTLP_LOGS_ENDPOINT', nil))
+      assert_nil(ENV.fetch('SW_APM_COLLECTOR', nil))
```

Comment:
Expected behaviour:

> The endpoint for sampler’s getSettings call is taken from SW APM config for collector, it should ONLY be the apm.collector.*.*.solarwinds.com or collector.appoptics.com endpoint since getSettings is not currently supported by the otel.collector.* endpoint. If SW APM config for collector is not set, fallback value is apm.collector.na-01.cloud.solarwinds.com.

so assertion should be `SW_APM_COLLECTOR` == `apm.collector.na-01.cloud.solarwinds.com`

Reply:
Updated